### PR TITLE
Rails Budget App

### PR DESCRIPTION
**Pull request by okoyecharles**
## Description ✍🏽
In this branch, I attempt to merge the required changes requested by microverse. I created a budget management application which I call **THE WALLET INSPECTOR** (_I know... I'm not the best at naming_).

The theme I chose is an application that could manage users' payments (the entity) under varying categories (the group).

Screenshot 2022-11-15 at 14 53 29

### Video Link  📸
Here is a [5min video link](https://www.loom.com/share/be262b368dde4fd8b896ab28dfe390bd) containing a more well-detailed explanation of the project and a demo of the project features.

- [x] Use ruby gems as a software package system.
- [x] Use controllers to handle requests and render empty views.
- [x] Use params from browser requests in a safe way.
- [x] Use preprocessed HTML files with embedded Ruby code.
- [x] Use layouts and templates for shared content.
- [x] Use database migration files to maintain database schema.
- [x] Use validations for models.
- [x] Secure app from n+1 problems.
- [x] Write SQL queries with ActiveRecord.
- [x] Set up associations between models.
- [x] Build a web app that requires the user to log in.
- [x] Use devise gem for authentication.
- [x] Resolve errors

### Participants (1)
* @OkoyeCharles

### Comments
Feel free to leave any comments in the PR thread if something is not 100% clear.
### Optional Suggestions
If you notice a problem that is not crucial enough to stop the approval of this PR. However, you strongly recommend I take them into account. Please I would love it if you let me know
### Clarity
Please it would be a pleasure if you could be as clear and precise as possible, as we have only three review attempts.
Thanks for understanding.

---------
_If there are any issues you would like to discuss quickly, you could contact me on [Slack](https://app.slack.com/client/T47CT8XPG/D039GHCKW3F), or (occasionally) on [WhatsApp](https://wa.me/+2348032533432)_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,5 +20,81 @@ Things you may want to cover:
 * Services (job queues, cache servers, search engines, etc.)
 
 * Deployment instructions
+![](https://img.shields.io/badge/Microverse-blueviolet)
```
Comment (sja-thedude) [inline README.md:23]: - [ ] Kindly remove from lines 1 to 22. It's not required.

File: README.md [hunk 2]
```diff
@@ -20,5 +20,81 @@ Things you may want to cover:
...
 
+# Budget-app project
+
+> This is a ruby on rails mobile web application where you can manage your budget: you have a list of payments associated with a category, so that you can see how much money you spent and on what.
+
+# Video link
+Here is the [video link](https://www.loom.com/share/50871a7a78944e64bc78acc36f6356bc) for the project.
+
```
Comment (sja-thedude) [inline README.md:31]: - [ ] Kindly make sure to deploy the app and add a link in the readme.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @OkoyeCharles, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Correct Gitflow
✔️ All linters passing

## Required 
- [ ] Kindly show in the loom video a highlighted piece of code that the student finds interesting or is proud of making.
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (OkoyeCharles) [reply README.md:31]: https://www.loom.com/share/a126415c78d2459d8b70c738e04fe14a

Comment (sja-thedude) [reply README.md:31]: @OkoyeCharles You can deploy your development branch with Heroku. The document microverse provided for deploying will be helpful.

Comment (OkoyeCharles) [reply README.md:31]: Okay I will look into it

Comment (OkoyeCharles) [reply README.md:12]: @sja-thedude Thanks very much for highlighting the above changes as you have helped me move my project one step to perfection.
Once again than you

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @OkoyeCharles👋, 

### You are doing a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Awesome UI design. ✔️



 ## Required Changes ♻️

- [ ] As requested by the previous reviewer, Please make sure the presentation video includes highlighted piece of code that the student finds interesting or is proud of making (You should re-record the presentation video and update it in the README file).

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by OkoyeCharles** [comment]
@thecodechaser I did include it... I added it in the README.md (development) as she requested.

**Comment by thecodechaser** [comment]
@OkoyeCharles In the video, I can't see you are showing the code. Please make sure to highlight some pieces of code.

**Comment by OkoyeCharles** [comment]
Okay sure I will do that

**Pull request merged**